Build fails due to conflicting ToroidCurrents module definitions

### Description

Following [this issue](https://github.com/ElmerCSC/elmerfem/issues/489), when building Elmer FEM, the build process fails with the following error:

```
[1913/2375] Generating Fortran dyndep file fem/tests/mgdyn_lamstack_lowfreq_transient/CMakeFiles/mgdyn_lamstack_lowfreq_transient_currents.dir/Fortran.dd
ninja: build stopped: multiple rules generate fmodules/toroidcurrents.mod.
```

This error occurs because there are multiple definitions of the `ToroidCurrents` module in different test directories.

### Files affected

The `ToroidCurrents` module is defined in the following files:

1. `fem/tests/mgdyn_anisotropic_cond/currents.f90`
2. `fem/tests/mgdyn_lamstack_lowfreq_transient/currents.f90`

### Steps to reproduce

1. Clone the Elmer FEM repository

    ```bash
    git clone https://github.com/ElmerCSC/elmerfem
    cd elmerfem
    git checkout adafcd2b7fcc4c5f804d2fcd2e9de0ba4350f630
    ```
2. Create a build directory and configure the build:
    ```bash
    mkdir build
    cd build
    cmake -D WITH_MPI=OFF ..
    ```
3. Build the project:
    ```bash
    cmake --build .
    ```

### Expected behavior

The build should complete successfully, properly handling the multiple definitions of the `ToroidCurrents` module.

### Actual behavior

The build fails due to conflicting rules for generating `fmodules/toroidcurrents.mod`.

### Possible solutions

1. Rename one of the `ToroidCurrents` modules to avoid the conflict.
2. Modify the CMake configuration to handle multiple definitions of the same module across different test directories.
3. Consolidate the `ToroidCurrents` module into a shared location that can be used by multiple tests.

### Additional information

This issue may be related to how Fortran modules are handled in the CMake build system, particularly for test cases that may have similar or identical module definitions.

I just tried the latest commit `77bbe42cd96e213af77e4e83453db905e2232f8f` and it still throws the error:

```
[1912/2375] Generating Fortran dyndep file fem/tests/mgdyn_lamstack_lowfreq_transient/CMakeFiles/mgdyn_lamstack_lowfreq_transient_currents.dir/Fortran.dd
ninja: build stopped: multiple rules generate fmodules/toroidcurrents.mod.
```

However, disabling Nijna and building for GNU Make without parallelism with ([reerence](https://www.msys2.org/docs/cmake/)):

```bash
cmake -D WITH_MPI=OFF -G "MinGW Makefiles" ..
mingw32-make
```

succeeds!

**P.S.** I asked a follow-up question [here](https://discord.com/channels/792780131906617355/1260936217948590112/1262664280411607133) on Discord.

--- a/.github/workflows/build-windows-mingw.yaml
+++ b/.github/workflows/build-windows-mingw.yaml
@@ -65,7 +65,6 @@ jobs:
           mkdir ${GITHUB_WORKSPACE}/build
           cd ${GITHUB_WORKSPACE}/build
           cmake \
-            -G "MSYS Makefiles" \
             -DCMAKE_BUILD_TYPE="Release" \
             -DCMAKE_INSTALL_PREFIX="${GITHUB_WORKSPACE}/usr" \
             -DCPACK_BUNDLE_EXTRA_WINDOWS_DLLS=OFF \

--- a/fem/tests/mgdyn_anisotropic_cond/currents.f90
+++ b/fem/tests/mgdyn_anisotropic_cond/currents.f90
@@ -1,18 +1,18 @@
 
-module ToroidCurrents
-  contains 
+module ToroidCurrents1
+  contains
 
     function rho(r, r0, y)
-      USE Types 
+      USE Types
       real(kind=dp) :: rho, r, r0, y
-      
+
        rho = sqrt((r-r0)**2+y**2)
     end function rho
 
     function currentInToroidR(r, r0, y, turns, I) result (curr)
-      USE Types 
+      USE Types
       real(kind=dp) :: r, r0, y, curr, rho1, rho2, turns, I
-   
+
       rho1 = rho(8d-3, r0, y)
       rho2 = rho(10d-3, r0, y)
 
@@ -21,9 +21,9 @@ module ToroidCurrents
     end function currentInToroidR
 
     function currentInToroidY(r, r0, y, turns, I) result (curr)
-      USE Types 
+      USE Types
       real(kind=dp) :: r, r0, y, curr, rho1, rho2, turns, I
-   
+
       rho1 = rho(8d-3, r0, y)
       rho2 = rho(10d-3, r0, y)
 
@@ -31,11 +31,11 @@ module ToroidCurrents
 
     end function currentInToroidY
 
-End module ToroidCurrents
+End module ToroidCurrents1
 
 FUNCTION currdens1( model, n, args) RESULT(curr)
   USE DefUtils
-  Use ToroidCurrents
+  Use ToroidCurrents1
   IMPLICIT None
   TYPE(Model_t) :: model
   INTEGER :: n
@@ -65,7 +65,7 @@ END FUNCTION currdens1
 
 FUNCTION currdens2( model, n, args) RESULT(curr)
   USE DefUtils
-  Use ToroidCurrents
+  Use ToroidCurrents1
   IMPLICIT None
   TYPE(Model_t) :: model
   INTEGER :: n
@@ -95,7 +95,7 @@ END FUNCTION currdens2
 
 FUNCTION currdens3( model, n, args) RESULT(curr)
   USE DefUtils
-  Use ToroidCurrents
+  Use ToroidCurrents1
   IMPLICIT None
   TYPE(Model_t) :: model
   INTEGER :: n
@@ -122,4 +122,3 @@ FUNCTION currdens3( model, n, args) RESULT(curr)
   curr = currentInToroidR(r, r0, y, turns, I) * cos(theta)
 
 END FUNCTION currdens3
-

--- a/fem/tests/mgdyn_lamstack_lowfreq_transient/currents.f90
+++ b/fem/tests/mgdyn_lamstack_lowfreq_transient/currents.f90
@@ -1,18 +1,18 @@
 
-module ToroidCurrents
-  contains 
+module ToroidCurrents2
+  contains
 
     function rho(r, r0, y)
-      USE Types 
+      USE Types
       real(kind=dp) :: rho, r, r0, y
-      
+
        rho = sqrt((r-r0)**2+y**2)
     end function rho
 
     function currentInToroidR(r, r0, y, turns, I) result (curr)
-      USE Types 
+      USE Types
       real(kind=dp) :: r, r0, y, curr, rho1, rho2, turns, I
-   
+
       rho1 = rho(8d-3, r0, y)
       rho2 = rho(10d-3, r0, y)
 
@@ -21,9 +21,9 @@ module ToroidCurrents
     end function currentInToroidR
 
     function currentInToroidY(r, r0, y, turns, I) result (curr)
-      USE Types 
+      USE Types
       real(kind=dp) :: r, r0, y, curr, rho1, rho2, turns, I
-   
+
       rho1 = rho(8d-3, r0, y)
       rho2 = rho(10d-3, r0, y)
 
@@ -31,11 +31,11 @@ module ToroidCurrents
 
     end function currentInToroidY
 
-End module ToroidCurrents
+End module ToroidCurrents2
 
 FUNCTION currdens1( model, n, args) RESULT(curr)
   USE DefUtils
-  Use ToroidCurrents
+  Use ToroidCurrents2
   IMPLICIT None
   TYPE(Model_t) :: model
   INTEGER :: n
@@ -65,7 +65,7 @@ END FUNCTION currdens1
 
 FUNCTION currdens2( model, n, args) RESULT(curr)
   USE DefUtils
-  Use ToroidCurrents
+  Use ToroidCurrents2
   IMPLICIT None
   TYPE(Model_t) :: model
   INTEGER :: n
@@ -95,7 +95,7 @@ END FUNCTION currdens2
 
 FUNCTION currdens3( model, n, args) RESULT(curr)
   USE DefUtils
-  Use ToroidCurrents
+  Use ToroidCurrents2
   IMPLICIT None
   TYPE(Model_t) :: model
   INTEGER :: n
@@ -122,5 +122,3 @@ FUNCTION currdens3( model, n, args) RESULT(curr)
   curr = currentInToroidR(r, r0, y, turns, I) * cos(theta)
 
 END FUNCTION currdens3
-
-

